test/index.js

```diff
@@ -1,42 +1,35 @@
 const assert = require('assert');
+const sinon = require('sinon');
 
+const modulePath = '../src/index';
 
 /* eslint-disable global-require, import/no-dynamic-require, no-console */
 describe('entry point', () => {
   beforeEach(() => {
-    this.consoleKeep = console.error;
+    delete require.cache[require.resolve(modulePath)];
   });
 
   afterEach(() => {
-    console.error = this.consoleKeep;
+    delete require.cache[require.resolve(modulePath)];
   });
 
   it('should return a function', () => {
-    assert(typeof require('../src/index'), 'function');
+    assert(typeof require(modulePath), 'function');
   });
 
-  it('should execute without an exception (no params) but print a warning message', (done) => {
-    console.error = (e) => {
-      assert.equal(e, 'Dial Once boot module should be initilised before used without config.');
-      done();
-    };
-    require('../src/index')();
+  it('should execute without an exception (no params) but print a warning message', () => {
+    const spy = sinon.spy(console.error);
+    console.error = spy;
```

Comment:
if you use const spy = sinon.spy(console, 'error'); on up line I think you won't ned this line.